src/core/packet.lua

```diff
@@ -72,6 +72,13 @@ function shiftleft (p, bytes)
    p.length = p.length - bytes
 end
 
+-- Move packet data to the right. This leaves length bytes of data
+-- at the beginning of the packet.
+function shiftright (p, bytes)
+   C.memmove(p.data + bytes, p.data, p.length)
```

Comment:
I think there has to be an `assert` here to check for overflow. See `packet.prepend`.